packages/core/interfaces/src/index.ts

```diff
@@ -158,6 +165,60 @@ export interface UseLocale
   loading: ComputedProperty<boolean>;
 }
 
+export interface ProductGetters<PRODUCT, PRODUCT_FILTER> {
+  getName: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<string>;
+  getSlug: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<string>;
+  getPrice: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<number | null>;
+  getGallery: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<UiMediaGalleryItem[]>;
+  getVariants: (products: PRODUCT[] | Readonly<PRODUCT[]>, filters?: PRODUCT_FILTER) =>
+    ComputedProperty<PRODUCT[]> | Readonly<PRODUCT[]>;
+  getAttributes: (product: PRODUCT[] | Readonly<PRODUCT[]>, filters?: Array<string>) =>
+    ComputedProperty<Array<AgnosticProductAttribute>>;
+  getDescription: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<any>;
+  getCategories: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<string[]>;
+  getId: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<number>;
+}
+
+export interface CartGetters<CART, PRODUCT> {
+  getProducts: (cart: CART) => ComputedProperty<PRODUCT[]>;
+  getProductName: (product: PRODUCT) => ComputedProperty<string>;
+  getProductImage: (product: PRODUCT) => ComputedProperty<string>;
+  getProductPrice: (product: PRODUCT) => ComputedProperty<string>;
+  getProductQty: (product: PRODUCT) => ComputedProperty<string>;
+  getProductAttributes: (product: PRODUCT, filterByAttributeName?: Array<string>) =>
+    ComputedProperty<Record<string, string | AgnosticProductAttribute>>;
+  getProductSku: (product: PRODUCT) => ComputedProperty<string>;
+  getTotals: (cart: CART) => ComputedProperty<AgnosticTotals>;
+  getShippingPrice: (cart: CART) => ComputedProperty<number>;
+  getTotalItems: (cart: CART) => ComputedProperty<number>;
+}
+
+export interface CategoryGetters<CATEGORY, PRODUCTS> {
+  getProducts: (category: CATEGORY, options: any) => ComputedProperty<PRODUCTS>;
+  getTree: (category: CATEGORY) => ComputedProperty<UiCategory | null>;
+  [getterName: string]: (element: CATEGORY, ...args: any) => any;
+}
+
+export interface UserGetters<USER> {
+  getFirstName: (customer: USER | Ref<USER>) => ComputedProperty<string>;
+  getLastName: (customer: USER | Ref<USER>) => ComputedProperty<string>;
+  getFullName: (customer: USER | Ref<USER>) => ComputedProperty<string>;
+}
+
+export interface CheckoutGetters<SHIPPING_METHODS> {
+  getShippingMethodId: (shippingMethod: SHIPPING_METHODS) => ComputedProperty<string>;
+  getShippingMethodName: (shippingMethod: SHIPPING_METHODS) => ComputedProperty<string>;
+  getShippingMethodDescription: (shippingMethod: SHIPPING_METHODS) => ComputedProperty<string>;
+  getShippingMethodPrice: (shippingMethod: SHIPPING_METHODS) => ComputedProperty<number>;
+}
+
+export interface UserOrderGetters<ORDER> {
+  getDate: (order: ORDER) => ComputedProperty<string>;
```

Comment:
`getItems` to get the ordered lines of products/items (including things like totals have - for instane, shipping price line)